service/service.go

```diff
@@ -135,39 +120,10 @@ func dial(target string, cert *x509.Certificate, insecureAllowed bool) (*grpc.Cl
 	))
 }
 
-func newCert(host string) (*x509.Certificate, error) {
-	cert, err := certFromHost(host)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-
-	if err := cacheRawCert(host, cert.Raw); err != nil {
-		return nil, errors.Trace(err)
-	}
-
-	return cert, nil
-}
-
 func credsFromCert(cert *x509.Certificate) (credentials.TransportCredentials, error) {
```

Comment:
the pool is never used, and the cert is always nil – this function doesn't need to exist